src/taskgraph/util/schema.py

```diff
@@ -17,52 +16,39 @@ def validate_schema(schema, obj, msg_prefix):
     """
     Validate that object satisfies schema.  If not, generate a useful exception
     beginning with msg_prefix.
+
+    Args:
+        schema: A msgspec.Struct type (including Schema subclasses)
+        obj: Object to validate
+        msg_prefix: Prefix for error messages
     """
     if taskgraph.fast:
         return
-    try:
-        schema(obj)
-    except voluptuous.MultipleInvalid as exc:
-        msg = [msg_prefix]
-        for error in exc.errors:
-            msg.append(str(error))
-        raise Exception("\n".join(msg) + "\n" + pprint.pformat(obj))
 
-
-def optionally_keyed_by(*arguments):
+    try:
+        if isinstance(schema, type) and issubclass(schema, Schema):
+            # Use the validate class method for Schema subclasses
+            schema.validate(obj)
+        elif isinstance(schema, type) and issubclass(schema, msgspec.Struct):
+            msgspec.convert(obj, schema)
```

Comment:
Do we need to support this use case? I'm not opposed, but if we never call `validate_schema` with a Struct object, then I'd mildly prefer we remove this code path. If we do call it, then this is totally fine.